.github/workflows/hpu-gaudi2.yml

```diff
@@ -39,7 +39,7 @@ jobs:
     # The type of runner that the job will run on
     runs-on: [self-hosted, intel, gaudi2]
     container:
-      image: vault.habana.ai/gaudi-docker/1.17.0/ubuntu22.04/habanalabs/pytorch-installer-2.3.1:latest
+      image: vault.habana.ai/gaudi-docker/1.18.0/ubuntu22.04/habanalabs/pytorch-installer-2.4.0:latest
```

Comment:
FYI @raza-sikander it appears that this test is running too long and timing out, do you know if this version is expected to be slower or how this could impact the test speed?

Reply:
The tests shouldnt be impacted. Let me check.

Reply:
It appears to work fine with 1.17.1 with either torch 2.3.1 or 2.4, it seems we might need to update the underlying software on the node itself.

Reply:
Yes, On local machine with latest image it works fine. On checking both hl-smi difference we can see firmware differences.

Reply:
It appears to work once we update the underlying node's firmware to 1.18 (from 1.17).  I've done that and will merge this PR shortly, thanks!